frontend/src/metabase/query_builder/components/AlertModals.jsx

```diff
@@ -109,16 +109,9 @@ export class CreateAlertModalContent extends Component {
   onAlertChange = alert => this.setState({ alert });
 
   onCreateAlert = async () => {
-    const {
-      question,
-      createAlert,
-      apiUpdateQuestion,
-      updateUrl,
-      onAlertCreated,
-    } = this.props;
+    const { question, createAlert, updateUrl, onAlertCreated } = this.props;
     const { alert } = this.state;
 
-    await apiUpdateQuestion(question);
```

Comment:
I'm no the author of this change, and I believe I've just moved this code from a different place. AFAIK this line couldn't be removed because you might want to create an alert for an unsaved question.

Reply:
that's interesting, I'll take a look.

Reply:
when I click create Alert on an unsaved question, It prompts me to save it first.

So I guess it's ok to remove this?

Reply:
I don't know all the cases where this modal is used. I'm not the author or owner of this feature, I can't tell for sure.

Reply:
Do on your own risk I guess.

Reply:
TLDR: I'll move forward with deleting that line.

Okay I chased the origin of this and it's from this [PR](https://github.com/metabase/metabase/pull/6278).
The main author is no longer working with us but some how @iethree is one of the contributor? 

It was implemented 5 years ago so I'd not hope anyone remember the background, but I found this [comment](https://github.com/metabase/metabase/blob/1a391c5e55706a7f506715d663b9efb429af362b/frontend/src/metabase/query_builder/components/AlertModals.jsx#L111), I guess the author's intention here is to save the qs before creating an alert if it's in edit mode ( or maybe it was used for dashboard back then hence the x/y axes )

I've double checked with our current behavior: if we create an alert if a question is either not save, in edit mode, we'll prompt user to save the Question first.

So I believe remove that line is totally safe.

Reply:
haha. there was a former ryan@metabase.com (4 years ago) and somehow I got credit for all his commits :grin: I probably won't be much help here.